python_modules/dagster/dagster/cli/api.py

```diff
@@ -84,9 +90,20 @@ def _execute_run_command_body(recon_pipeline, pipeline_run_id, instance, write_s
         EngineEventData.in_process(pid, marker_end="cli_api_subprocess_init"),
     )
 
+    run_worker_failed = 0
+
     try:
-        for event in core_execute_run(recon_pipeline, pipeline_run, instance):
+        for event in core_execute_run(
+            recon_pipeline,
+            pipeline_run,
+            instance,
+        ):
             write_stream_fn(event)
+            if event.event_type == DagsterEventType.PIPELINE_FAILURE:
+                run_worker_failed = True
+    except:
+        # relies on core_execute_run writing failures to the event log before raising
+        run_worker_failed = True
```

Comment:
tried making core_execute_run always raise and the (small number of) callsites now have to explicitly catch.